plugins/degu.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Plugins for using Degu elements.
+ */
+
+import {NunjucksTemplateEngine, Pod} from '@amagaki/amagaki';
+
+import nunjucks from 'nunjucks';
+
+export class DeguPlugin {
```

Comment:
Interested in feedback on this approach; as we get more degu elements, we can create the equivalent of Nunjucks macros here. I like using `.ts` instead of `.njk` for these foundational elements a little more because it helps with correctness and testability. (This file could easily be a `.njk` macro instead, except for those two points.)

This "degu nunjucks plugin" can just help with outputting the correct HTML markup for the foundational degu elements. We can eventually merge this type of thing into an `@amagaki/amagaki-plugin-degu` (for example) Amagaki plugin to help with maintainability across projects.

@uxder @stevenle

Reply:
Oh, one more thing. If you browse the repo at HEAD/`main` now you can see how minimal the project is getting and how we're moving towards having less boilerplate, which I think is a good thing. :)

Reply:
One problem with this is that there's no longer any HTML sanitization. You have a XSS issue here for example, what if `alt` comes from something that's user-entered and they have a quote in the alt text? 

As long as you have a way of sanitizing the input, this approach seems fine, but security should be a priority.

Reply:
Security might be a good point but apart from that, +1 to this style.  I definitely like how it is more testable and readable.

Reply:
Just want to note re: commit `da225ab`, rendering a njk file that calls into a ts plugin that calls njk feels like an anti-pattern. You can sanitize your inputs by using some html encoder and that would probably be the easiest without this extra overhead. That said, if this is the pattern moving forward, every developer will need to have security in mind since we can't just assume that there are no bad actors trying to abuse the system.

Reply:
Thanks both - these are really good points and I didn't consider the security aspect. The tags weren't meant to be used with end-user supplied content but after reading this thread I fully agree that we can't predict where the content originates so we should provide some guarantee that it behaves as expected especially when used with Nunjucks.

`da225ab` was meant to go in another branch so we can compare two approaches (it's reverted in this branch now):
https://github.com/blinkk/amagaki-starter/compare/feature/degu...feature/degu-njk - I agree I certainly like the lighter approach that doesn't involve `njk` creation within a `.ts` file.

Regardless I've updated this PR with an improved approach that uses the original template literals but with an added `html` tag function that uses Nunjucks' `SafeString` and `escape`. `escape` runs on all the inputs.

What do you both think of this? I certainly prefer this to straight up Nunjucks macros for shared code. Eventually I think we'll move this to its own package or part of a different plugin rather than it being boilerplate in the starter.

Reply:
This new approach seems better and can be more easily copied from plugin-to-plugin without much overhead. 

My only other concern is dependency management on the custom elements side, this one for example declares `<degu-image>` as a dependency which is hardcoded in the main.ts, but as noted you'll eventually need to include `<degu-video>` as well. I think ideally there'd be a way for the plugins to say "if you're using this plugin, automatically inject these custom element dependencies into the page" somehow. Something to think about if/when this is moved to its own package.

Reply:
Great! Yeah, this will be something for us to think about. I think I'd like to get this approach "blessed" in the starter first, and then likely move the string builder method to the PageBuilder plugin, along with other things.

I generally think it's good to have the user specify exactly where they want to load the custom elements (rather than us do it for them) as the more we do for them the more like a "frontend framework" this becomes. That may not be so bad, but I think providing the instruction/flexibility to do the custom element loading in `main.ts` as a starting point is good (at least for now).